tests/theme/dark-mode.spec.tsx

```diff
@@ -0,0 +1,191 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import styled from "styled-components";
+import {
+    Colour,
+    DSThemeProvider,
+    LifeSGTheme,
+    useDSTheme,
+} from "../../src/theme";
+
+// Test component that uses theme colors
+const TestComponent = styled.div`
+    background: ${Colour["bg"]};
+    color: ${Colour["text"]};
+`;
+
+// Component to test the useDSTheme hook
+const ThemeModeDisplay = () => {
+    const { colourMode } = useDSTheme();
+    return <div data-testid="theme-mode">{colourMode}</div>;
+};
+
+describe("Dark Mode Theme", () => {
```

Comment:
we could assert that the tokens are mapped correctly in styled-components

for example we could render the test component like so:

```
const ThemeModeDisplay = () => {
    const { colourMode } = useDSTheme();
    return (
        <div>
            <div data-testid="theme-mode">{colourMode}</div>
            <div>
                <TestComponent data-testid="theme-component" />
            </div>
        </div>
    );
};
```

then the first test case can have these assertions
```
        expect(getByTestId("theme-mode")).toHaveTextContent("light");
        expect(getByTestId("theme-component")).toHaveStyle({
            color: "#282828",
            background: "#ffffff",
        });
```